### PR TITLE
kv: add API for kv.Client

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

File: kv/kv.go [hunk 1]
```diff
@@ -76,6 +78,53 @@ type Transaction interface {
...
+	SupportRequestType(reqType, subType int64) bool
+}
+
+// ReqTypes.
+const (
+	ReqTypeSelect = 101
+	ReqTypeIndex  = 102
+)
+
+// KeyRange represent a range where StartKey <= key < EndKey.
```
Comment (zimulala) [inline kv/kv.go]: represent -> represents ?

**Comment by coocood** [comment]
@zimulala @shenli PTAL

**Comment by coocood** [comment]
@ngaut PTAL

**Comment by ngaut** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**